gc/base/MemorySubSpace.cpp

```diff
@@ -919,17 +919,51 @@ MM_MemorySubSpace::systemGarbageCollect(MM_EnvironmentBase* env, uint32_t gcCode
 	}
 
 	if (_collector && _usesGlobalCollector) {
-		/* TODO: This is bogus for multiple memory spaces - should ask the space, not the heap */
-		_extensions->heap->getResizeStats()->setFreeBytesAtSystemGCStart(getApproximateActiveFreeMemorySize());
+#if defined(OMR_GC_IDLE_HEAP_MANAGER)
+		bool invokeGC = true;
+		if (J9MMCONSTANT_EXPLICIT_GC_IDLE_GC == gcCode ) {
+			MM_MemorySpace* defaultMemorySpace = _extensions->heap->getDefaultMemorySpace();
+			uintptr_t freeMemorySize = defaultMemorySpace->getApproximateActiveFreeMemorySize(MEMORY_TYPE_OLD);
+			uintptr_t actvMemorySize = defaultMemorySpace->getActiveMemorySize(MEMORY_TYPE_OLD);
+			uintptr_t previousMemorySize;
```

Comment:
local variable should be initialized... Suggestion: There is might be a case when using of arithmetic if is reasonable